src/vm/core.zig

```diff
@@ -147,29 +149,32 @@ pub const CairoVM = struct {
         // Compute the destination address and get value from the memory.
         const dst_addr = try self.run_context.compute_dst_addr(instruction);
         const dst = try self.segments.memory.get(dst_addr);
-        _ = dst;
 
         // Compute the OP 0 address and get value from the memory.
         const op_0_addr = try self.run_context.compute_op_0_addr(instruction);
         // Here we use `catch null` because we want op_0_op to be optional since it's not always used.
-        const op_0_op = self.segments.memory.get(op_0_addr) catch null;
+        const op_0_op = try self.segments.memory.get(op_0_addr);
 
         // Compute the OP 1 address and get value from the memory.
         const op_1_addr = try self.run_context.compute_op_1_addr(
             instruction,
             op_0_op,
         );
         const op_1_op = try self.segments.memory.get(op_1_addr);
-        _ = op_1_op;
 
-        return .{
-            .dst = relocatable.fromU64(0),
-            .res = relocatable.fromU64(0),
-            .op_0 = relocatable.fromU64(0),
-            .op_1 = relocatable.fromU64(0),
-            .dst_addr = relocatable.fromU64(0),
-            .op_0_addr = relocatable.fromU64(0),
-            .op_1_addr = relocatable.fromU64(0),
+        const res = try computeRes(instruction, op_0_op, op_1_op);
+
+        // Deduce the operands if they haven't been successfully retrieved from memory.
+        // TODO: Implement this.
+
+        return OperandsResult{
```

Comment:
I believe this works?

```suggestion
        return .{
```

Reply:
It does! Magic